### DISCUSSION TITLE
Linux Support

### DISCUSSION THREAD

**Discussion by rosie-at-pieces** [pieces-app/support#14] [Announcements]
Pieces for Developers currently only supports ubuntu and ubuntu-based distributions. Other OSs may work but we can't guarantee the quality of experience.

We would love to hear from our community what distributions you are using and how Pieces is performing on it - please leave a comment below with your experience!

**Comment by rosie-at-pieces** [comment]
@nathan-courtney-pieces feel free to add some more details on the versions we know work well

**Reply by nathan-courtney-pieces** [reply]
22.04 and Up currently

**Comment by peaklabs-dev** [comment]
I use Debian and Red Hat Linux mainly

**Comment by priyankarpal** [comment]
I am currently experiencing issues with Kali Linux (Xfce). Below is the error log.

![Screenshot_2024-02-23_09_14_39](https://github.com/pieces-app/support/assets/88102392/9d535569-1a44-4d37-ac04-327bd8338554)

```
error: cannot communicate with server: Post "http://localhost/v2/snaps/pieces-os": dial unix /run/snapd.socket: connect: no such file or directory
```

**Reply by dealeros** [reply]
Are you running this in distrobox/container? Unfortunately it won't work this way. I tried this too.

**Comment by ketodin** [comment]
I'm on an Arch Linux based distro, EndeavourOS and its now working fine since the proxy update. Firefox, VSCode extension, and Pieces for developpers work seamlessly thanks to Pieces-OS.

**Comment by rosie-at-pieces** [comment]
For all our Arch users - here's some advice for getting Pieces to work. Remember we don't officially support Arch at this time, but based on feedback and our own experience, you can likely make it work reasonably well.

Make sure you have snapd set up and running properly, and that you don't attempt to run Pieces with elevated privileges. 

The other biggest factor is your graphics drivers, which could vary in different ways. You may need to mess around with graphics drivers to see which one works with Pieces on their machine. 

There's a lot of unknowns about how individual users set up their machines. For example, someone could have unintentionally removed read and write privileges to their Documents folder from their user account because they have a habit of running everything on their machine with elevated privileges.

Another example is a lot of Arch users will choose a desktop environment like GNOME or KDE. We only support GNOME (ubuntu default).

Since Arch is just a terminal based OS, you have to explicitly configure a desktop environment if you want to "see" anything on your monitor other than a terminal. Supporting Arch out of the box is really tough for us because flutter is dependent on certain graphical libraries that don't exist out of the box.

I hope this is insightful! Feel free to add your experience here as well.

**Comment by vyrx-dev** [comment]
Before when I was using pieces, it was displaying 'pieces-os is not up to date,' but now it's resolved. However, when I try to link it to my GitHub Google account, it fails to redirect me to the browser for signing in. Consequently, I'm unable to utilize the backup and restore feature.

Anyone know how to fix it

I am using vanilla arch with hyprland wm

**Reply by bellomondo** [reply]
> Before when I was using pieces, it was displaying 'pieces-os is not up to date,' ***but now it's resolved***.

How did you resolve it? In my case it's the other way around, Worked flawlessly and then started showing me this error for some reason...

I'm using EndeavourOS (GNOME).

I would love to see an AUR package, to stay on topic.

**Comment by rosie-at-pieces** [comment]
Heard a request for flatpak today!

**Reply by Obsidian-Jackal** [reply]
> Heard a request for flatpak today!

I would love to have flatpak support, so I can use it on Nitrux OS and other non-Ubuntu based OSes.

**Comment by vyrx-dev** [comment]
Yes it's far better than snap

**Comment by Sherwin-14** [comment]
I'd love flatpak support too! Most RHEL derivatives require disabling SELinux in order to install snapd.

**Comment by justchokingaround** [comment]
would love to see NixOS support

**Comment by progressEdd** [comment]
I would also like to see a flatpak support. I use fedora for my personal computer

**Reply by dealeros** [reply]
Yup, flatpak would be nice in openSuse too.

**Comment by fritogotlayed** [comment]
I too would like to see flatpak support. In my case I'm running Pop_OS! and the snap store is disabled by default. So while I technically am ubuntu-based this is a bit of a blocker for anyone that uses a similar ubuntu-base that has snap disabled by default or other distros that use flatpak over snap.

**Comment by pcause** [comment]
Fedora please!

**Comment by rajbaratht** [comment]
I use Arch linux. Please support flatpak.

**Comment by special-k-anon** [comment]
Flatpak or AppImage would be awesome. Also, I've noticed that the Windows version has new updates all the time. Is the snap going to see any updates in the near future? I've had it for at least 6 months and I don't think I've ever seen an update.

**Comment by CodyErekson** [comment]
I refuse to use Snaps, I abandoned Ubuntu because of them. Please offer something else. I was excited to try this app until I learned it required Snaps. Please provide a deb.

**Comment by BeZazz** [comment]
It is running fine (any issues are listed as bugs) for me on 
Fedora 41
CPU AMD A8-3870 APU
GPU NVIDIA GeForce GTX 1080
